core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -120,17 +121,20 @@ class TimeIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writable:
       // because that could happen in the following two scenarios:
       // 1. A log segment is closed.
       // 2. LogSegment.onBecomeInactiveSegment() is called when an active log segment is rolled.
-      if (_entries != 0 && offset < lastEntry.offset)
-        throw new InvalidOffsetException(s"Attempt to append an offset ($offset) to slot ${_entries} no larger than" +
-          s" the last offset appended (${lastEntry.offset}) to ${file.getAbsolutePath}.")
-      if (_entries != 0 && timestamp < lastEntry.timestamp)
-        throw new IllegalStateException(s"Attempt to append a timestamp ($timestamp) to slot ${_entries} no larger" +
-          s" than the last timestamp appended (${lastEntry.timestamp}) to ${file.getAbsolutePath}.")
+      if (_entries != 0) {
+        if (offset < lastEntry.offset)
```

Comment:
maybe we can ignore this optimization to reduce the possible merge conflicts in the future. It is up to you.

Reply:
I feel that this provides a performance benefit. So I prefer keeping it. To maintain consistency, I will try to push it upstream, as well.